### PR TITLE
FINERACT-2098: Currency data needed in loan transaction for UI

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

For displaying the Input amount control is required to have the Currency data in the Loan Transaction template 

[FINERACT-2098](https://issues.apache.org/jira/browse/FINERACT-2098)

Screenshot 2024-06-17 at 8 36 20 a m

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -539,7 +539,9 @@ public LoanTransactionData retrieveNewClosureDetails() {
     @Override
     public LoanApprovalData retrieveApprovalTemplate(final Long loanId) {
         final Loan loan = this.loanRepositoryWrapper.findOneWithNotFoundDetection(loanId, true);
-        return new LoanApprovalData(loan.getProposedPrincipal(), DateUtils.getBusinessLocalDate(), loan.getNetDisbursalAmount());
+        final CurrencyData currencyData = new CurrencyData(loan.getCurrencyCode());
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java]: Please send the full currency data, just as it sent for repayment template call.
`return new CurrencyData(currencyCode, currencyName, currencyDigits, inMultiplesOf, currencyDisplaySymbol, currencyNameCode);`

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly see my review!
Please dont forget to cover with proper test cases the changed behaviour! ;)

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java (hunk 1 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java]: Done! PR updated

**Comment by alberto-art3ch** [comment]
> Kindly see my review! Please dont forget to cover with proper test cases the changed behaviour! ;)

Done! PR updated

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalIdSupportIntegrationTest.java [hunk 2]
```diff
@@ -900,6 +900,8 @@ public void loan() {
             assertEquals(actualDate, loanApprovalResult.getApprovalDate());
             assertEquals(1000.0, loanApprovalResult.getApprovalAmount());
             assertEquals(1000.0, loanApprovalResult.getNetDisbursalAmount());
+            assertNotNull(loanApprovalResult.getCurrency());
```
Comment (adamsaghy) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalIdSupportIntegrationTest.java:904]: Would you mind to rewrite to check the value of the currency code from the loan approval response is matching with the loan product currency?

Also you have added the currency for disbursal template as well. Please write test for that one as well!

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly check my review!

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalIdSupportIntegrationTest.java (hunk 2 above)
Comment (alberto-art3ch) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalIdSupportIntegrationTest.java:904]: Done! PR updated with the cnanges requested

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**